packages/trader/src/AppV2/Components/ContractDetailsFooter/__tests__/contract-details-footer.spec.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { useStore } from '@deriv/stores';
+import { getCardLabels, isValidToSell, isValidToCancel, isMultiplierContract } from '@deriv/shared';
+import ContractDetailsFooter from '../contract-details-footer';
+import { getRemainingTime } from 'AppV2/Utils/helper';
+import userEvent from '@testing-library/user-event';
+
+jest.mock('@deriv/stores', () => ({
+    useStore: jest.fn(),
+}));
+jest.mock('@deriv/shared', () => ({
+    getCardLabels: jest.fn(),
+    isValidToSell: jest.fn(),
+    isValidToCancel: jest.fn(),
+    isMultiplierContract: jest.fn(),
+}));
+jest.mock('AppV2/Utils/helper', () => ({
+    getRemainingTime: jest.fn(),
+}));
+
+const mockContractInfo = {
+    bid_price: 100,
+    currency: 'USD',
+    contract_id: 1,
+    profit: 10,
+    contract_type: 'non-multiplier',
+};
+
+describe('ContractDetailsFooter', () => {
+    let mockStore: any;
+
+    beforeEach(() => {
+        mockStore = {
+            contract_replay: {
+                onClickCancel: jest.fn(),
+                onClickSell: jest.fn(),
+                is_sell_requested: false,
+            },
+            common: {
+                server_time: new Date(),
+            },
+        };
+
+        (useStore as jest.Mock).mockImplementation(() => mockStore);
+        (getCardLabels as jest.Mock).mockImplementation(() => ({
```

Comment:
Why do we need to mock `getCardLabels`? It's just returning object with same strings, which you wrote

Reply:
fixed thanks